### PR TITLE
feat(Examples,CMSIS,PeriphDrivers): Add support for MAX32675 RevB

### PR DESCRIPTION
**Pull request by sihyung-maxim**
### Description

This PR adds support to the MAX32675 Revb.

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Comment by Jake-Carter** [comment]
@sihyung-maxim @lorne-maxim is our goal also to maintain full driver compatibility with Rev A hardware?

**Comment by sihyung-maxim** [comment]
No, there are certain differences with these new parts that it won't be easy to fully support both revisions. I believe the RevA hardware is out of production, and we aren't expected to support those parts soon. However, I've made it to where we'd still support the RevA hardware as best we can for now while making any RevB hardware changes a priority - if attempting to support both revs for a certain library is going to take longer than it's worth.

At the moment, it's important to get this out the door for the customers that are receiving the first batch of these parts.

File: Examples/MAX32675/I2S/main.c [hunk 1]
```diff
@@ -74,12 +76,24 @@ int main(void)
...
     mxc_i2s_req_t req;
+
     printf("\nI2S Transmission Example\n");
     printf("I2S Signals may be viewed on pins P0.8-P0.11.\n");
     printf("You may need to disconnect RX_SEL (JP5) and TX_SEL\n");
     printf("(JP6) in case no data is moving in and out of SDO/SDI.\n");
+    printf("\nCurrent revision of board: %x\n", MXC_GCR->revision);
+    if ((MXC_GCR->revision & 0x00F0) == 0x00B0) {
+        printf("I2S is not support on this MAX32675 revision.\n");
+        printf("End program now.");
```
Comment (lorne-maxim) [inline Examples/MAX32675/I2S/main.c:87]: not support**ed**
End**ing** program

File: Libraries/CMSIS/Device/Maxim/MAX32675/Source/system_max32675.c [hunk 2]
```diff
@@ -153,9 +153,11 @@ __weak void SystemInit(void)
     __DSB();
     __ISB();
 
-    /* Change system clock source to the main high-speed clock */
-    MXC_SYS_Clock_Select(MXC_SYS_CLOCK_IPO);
-    SystemCoreClockUpdate();
+    /* The default clock for the MAX32675 Revision B is the IBRO. Do not update. */
+    if ((MXC_GCR->revision & MXC_F_GCR_REVISION_REVISION) == 0x03B4) {
```
Comment (lorne-maxim) [inline Libraries/CMSIS/Device/Maxim/MAX32675/Source/system_max32675.c:157]: Should the reverse logic be used?  That is, only do the clock change if rev is not Rev B.

Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32675/Source/system_max32675.c]: Removed the entire if statement.

**Review by lorne-maxim — APPROVED** [review]

**Pull request merged**